torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -639,6 +691,9 @@ bool FunctionSignature::parse(PyObject* args, PyObject* kwargs, PyObject* dst[],
   }
 
   int i = 0;
+  if (self != nullptr && check_has_torch_function(self)) {
```

Comment:
Isn't this the slow check? Look at the old code:

```
      if (THPVariable_Check(obj)) {
        if (check_has_torch_function(obj)) {
          append_overloaded_arg(overloaded_args, obj);
        }
        return true;
      }
```

You need to do the fast check first (`THPVariable_Check`) before doing the slow check. I imagine this is at least some of the unaccounted for cost.

We should probably have documented this more clearly in `check_has_torch_function` (maybe call it `slow_check_has_torch_function` and written a Note about it.)

Reply:
Also, this seems to imply that all of the boilerplate `check_has_torch_function` calls are unnecessarily slowing things down.

Reply:
I seem to recall I checked this. Going back and verifying it, pretty much the first thing `check_has_torch_function` does is call `PyTorch_LookupSpecial`, and the first thing that does is an `THPVariable_CheckExact` call, which returns a `py::object()`. This is checked for `nullptr` later. If you want, I can add an explicit `THPVariable_CheckExact` call.